CreeDictionary/search_quality/analyze_results.py

```diff
@@ -0,0 +1,191 @@
+# The analyses in this file don’t touch the database, and might perform some
+# expensive operations, because they are meant for analyzing and comparing
+# outputs generated today using the latest everything to previously-generated
+# outputs using old versions of the code and old versions of the dictionary.
+
+from __future__ import annotations
+
+import gzip
+from json import load
+from os import PathLike
+from typing import cast, TypedDict, Optional, Sequence
+
+from API.schema import SerializedSearchResult
+from utils.types import FSTTag
+
+from . import DEFAULT_SAMPLE_FILE, SampleSearchResultsJson
+from .sample import load_sample_definition, SampleDefinition
+
+
+class DuplicateInfo(TypedDict):
+    total: int
+    unique: int
+
+
+class SingleQueryAnalysis(TypedDict, total=False):
+    time_taken_seconds: float
+    result_count: int
+    total_duplicate_count: int
+    unique_duplicate_count: int
+    topthree_score_percent: float
+    threeten_score_percent: float
+    word_1: str
+    word_1_rank: int | str
+    word_2: str
+    word_2_rank: int | str
+    word_3: str
+    word_3_rank: int | str
+
+
+Analysis = dict[str, Optional[SingleQueryAnalysis]]
+
+
+def count_results(results: list[SerializedSearchResult]):
+    return sum(1 if r["is_lemma"] else 2 for r in results)
+
+
+class OptionalPartialSerializedWordform(TypedDict, total=False):
```

Comment:
Why both `Optional` and `Partial`? Wouldn't it just be a partial (non-total) `SerializedWordform`?

Reply:
Good point.